### DISCUSSION TITLE
[Projects Beta] [Insights] Option to include/exclude archived issues for charts

### DISCUSSION THREAD

**Discussion by brendangadd** [community/community#13356] [Projects and Issues]
The way charts handle archived issues limits the effectiveness and flexibility of charts/insights. I think there are two main improvements that would be helpful:

* Handle archived issues consistently
* Give the user control over archived issue inclusion

### Handle archived issues consistently

The behaviour for including archived issues seems to be inconsistent right now.

When I look at the _Burn up / CFD_ default chart, all archived issues appear to be included in the visualization. In my case, that meant the chart wasn't useful to me (since it isn't configurable), as the data is completely dominated by completed archived issues. I had to create a duplicate chart with a filter to get useful data.

Seeing that archived issues are included, I tried to use that behaviour to make a chart that shows velocity over time (i.e. Sum of `Points` (number), `status:Done`, plotted per `Sprint` (iteration)). But as soon as I archived items from the previous sprint, they were excluded from the chart.

Now I don't know what behaviour to expect, or if I'm doing something wrong. I think the behaviour should be consistent: probably including archived issues by default. But I think it's even more important to be able to control this behaviour.

### Give the user control over archived issue inclusion

As described above, there's a need to include or exclude archived issues on a per visualization basis.

* I want to sometimes **exclude** archived issues so that my visualizations aren't dominated by hundreds of old completed issues
* I want to sometimes **include** archived issues so that I can visualize historical data

I think there are a couple ways this could be made available:

* Checkbox in the _Configure_ UI of the chart
* Something akin to an `is:archived` property we can use for filters (e.g. include by default, add `-` to hide)

### Note on workarounds

I'm guessing this issue can be mostly worked around by creating some sort of "Archived" status and then filtering on it to include in select visualizations and hide from basically all other Projects Beta views. But that essentially re-implements and replaces the true archival feature, which signals to me that this a feature worth adding.

**Comment by 700software** [comment]
Regarding removal from all charts I have no opinion.

Regarding *retroactive* removal from Time series charts e.g. Burn Up / CFD
created https://github.com/github/feedback/discussions/17280 as that's unusable at the moment.

**Comment by aeneasr** [comment]
Despite having now access to more items on the board (50k), the issue remains. We still have to remove old items from our views (aka archive) because it clogs up the views and makes the boards very slow. With archived items though, these dashboards are essentially useless (we'd love to see some stats but the stats shown are wrong because the majority is filtered out).

**Comment by SeaRoll** [comment]
Still an issue!

**Comment by slinna** [comment]
If a paid add-on is an option, I suggest taking a look at the [Analytics & Reports](https://screenful.com/github) by Screenful. It will keep the archived items in the stats so you can track your metrics over longer periods of time. Here's a [list of metrics you can track](https://screenful.com/guide/github/what-metrics-can-i-track-with-screenful-for-github) with Screenful. 

Available in the [GitHub Marketplace](https://github.com/marketplace/analytics-reports).
